### DISCUSSION TITLE
Possible to clip a water plane?

### DISCUSSION THREAD

**Discussion by UnknownTarget** [mozilla/hubs#3335] [Spoke Questions] [answered]
I want to use a water plane but I want my object partially submerged in it. However the object has a lot of odd shapes; for example, there's a bowl cut out in the middle of it. So the water plane clips into it. I'd like to hide the water plane from that area; possible?

**Comment by cvanmeurs** [comment]
It sounds like what you are wanting to do is a boolean subtraction.  It's possible to do in a 3D software such as Blender.

**Comment by UnknownTarget** [comment, marked as answer]
No, I would like to perform a boolean on the water plane include in Spoke.
It's possible to do in Blender but since the water plane has special
effects within Spoke, I'd like to be able to do it in there. It's a lot
easier than having to bake out a bunch of animated morph maps.

**Reply by cvanmeurs** [reply]
I gotcha.  Yeah unfortunately there are no boolean or cutout type objects in Spoke.  It's an interesting idea tho.  Jim is in fact making some new Spoke primitive shape kit pieces this week.  I'll ask around about the feasibility of something like this.

Looking at the "Simple Water" plane in spoke, it does look like it's just a combination of a couple morph targets and a scrolling texture or two.  Should be fairly simple to re-create using our "Loop Animation" and "UV Scroll" components in Blender.

**Comment by UnknownTarget** [comment]
Awesome, thanks!

Take a look at TinkerCAD for a good approach to handling simple shape
booleans. :)
